### PR TITLE
Retrieve children for all body areas

### PR DESCRIPTION
**Pull request by chrismytton**
Updates the code that fetches children for a body to look at all associated areas.

This fixes the issue that Northamptonshire reported where they could only see North Northamptonshire wards in the dashboard dropdown.

Fixes https://github.com/mysociety/societyworks/issues/3262

[skip changelog]

### CODE REVIEW

File: perllib/FixMyStreet/Cobrand/Default.pm [hunk 1]
```diff
@@ -814,12 +814,21 @@ cobrand's area_types_children type.
...
 
-    return FixMyStreet::MapIt::call('area/children', $area_id,
-        type => $self->area_types_children,
-        $all_generations ? (min_generation => 1) : (),
-    );
+    $area_ids = [ $area_ids ] unless ref $area_ids eq 'ARRAY';
+
+    my %all_children;
+
+    foreach my $area_id (@$area_ids) {
```
Comment (dracos) [inline perllib/FixMyStreet/Cobrand/Default.pm:823]: You can call children with an array ref of IDs (e.g. `call('area/children', $area_ids)`) and that should get you all the results you need with one call.

**Review by dracos — CHANGES_REQUESTED** [review]
Nice, this should also fix the list of wards on the All Reports page for Northants too :) One suggestion as to how to only make the one MapIt call is all, I think.

File: perllib/FixMyStreet/Cobrand/Default.pm (hunk 1 above)
Comment (chrismytton) [reply perllib/FixMyStreet/Cobrand/Default.pm:823]: I tried doing that but it wasn't returning any results. Spoke to Dave about it and it seems Mapit doesn't have code for looking up children for multiple areas. Unless I've missed something?

https://mapit.mysociety.org/areas/164185,164186/children?type=DIW,LBW,MTW,UTE,UTW,LGE,CED,COP&generation=42

(Not sure what's actually handling that route)

I did briefly start looking at changing Mapit, but seeing as there aren't many bodies with two (or more) areas this seemed like the simpler solution.

Comment (dracos) [reply perllib/FixMyStreet/Cobrand/Default.pm:823]: Oh, I'm just talking nonsense then, sorry!

**Review by dracos — APPROVED** [review]

**Pull request merged**